runtime/dart_isolate.cc

```diff
@@ -742,10 +742,29 @@ bool DartIsolate::RunFromLibrary(std::optional<std::string> library_name,
   auto entrypoint_handle = entrypoint.has_value() && !entrypoint.value().empty()
                                ? tonic::ToDart(entrypoint.value().c_str())
                                : tonic::ToDart("main");
+  auto entrypoint_args = tonic::ToDart(args);
   auto user_entrypoint_function =
       ::Dart_GetField(library_handle, entrypoint_handle);
 
-  auto entrypoint_args = tonic::ToDart(args);
+  // The Dart plugin registrant is a static with signature `void register()`
+  // within the class `_PluginRegistrant` generated by the Flutter tool.
+  //
+  // This method binds a plugin implementation to their platform
+  // interface based on the configuration of the app's pubpec.yaml, and the
+  // plugin's pubspec.yaml.
+  //
+  // Since this method may or may not be defined, check if the class is defined
+  // in the default library before calling the method.
+  Dart_Handle plugin_registrant =
+      ::Dart_GetClass(library_handle, tonic::ToDart("_PluginRegistrant"));
+
+  if (!Dart_IsError(plugin_registrant)) {
+    Dart_Handle result =
+        tonic::DartInvokeField(plugin_registrant, "register", {});
+    if (tonic::LogIfError(result)) {
+      FML_LOG(ERROR) << "Could not invoke the Dart plugin registrant.";
```

Comment:
Not sure this is right. It might have been invoked, but resulted in an unhandled exception. It depends on how `tonic::DartInvokeField` is written.

Reply:
Good point. I don't know what would happen if the method threw an exception. I think a test that that throws an exception in `register` and reasons about it is warranted. Do do this, we can use `Settings.unhandled_exception_callback` to listen for the error. Or, make sure the generated register method doesn't throw by adding a try-catch around its contents. I suppose, either way, this must be resilient to exceptions.

Reply:
This is blocked on the previous comment - throwing from the static method `void register()` requires a new Dart main file.

Reply:
We might need to put try/catch wrappers around each plugin's `registerWith` method. https://github.com/flutter/flutter/pull/78964 is doing this for Android.